container_test.go

```diff
@@ -1375,3 +1376,33 @@ func TestContainerHostname(t *testing.T) {
 		t.Errorf("expected output %q but received %q", expected, actual)
 	}
 }
+
+func TestContainerExtensions(t *testing.T) {
+	t.Parallel()
+
+	client, err := newClient(t, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := testContext()
+	defer cancel()
+	id := t.Name()
+
+	ext := gogotypes.Any{TypeUrl: "test.ext.url", Value: []byte("hello")}
+	container, err := client.NewContainer(ctx, id, WithContainerExtensions(ext))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cExt := container.Extensions()
+	if cExt == nil {
+		t.Fatal("got nil container extensions")
+	}
+	if cExt.TypeUrl != ext.TypeUrl {
+		t.Fatalf("got unexpected type url for extension: %s", cExt.TypeUrl)
+	}
+	if !bytes.Equal(cExt.Value, ext.Value) {
+		t.Fatalf("expected extension value %q, got: %q", ext.Value, cExt.Value)
+	}
```

Comment:
could you also add a a test where you load the container and check if the data is correctly sent back?

Also, don't forget to clean up after the test, this need a `container.Delete()` call.